source/Meadow.Core/Update/UpdateService.cs

```diff
@@ -200,7 +227,7 @@ private async void UpdateStateMachine()
             switch (State)
             {
                 case UpdateState.Disconnected:
-                    if (Config.UseAuthentication)
+                    if (Config.UseAuthentication && ShouldAuthenticate())
```

Comment:
Please let me know @ctacke if you think that I shouldn't add the `ShouldAuthenticate` method here.

I added it to avoid unnecessary authentication steps since it's quite common for poor cell connection to lose connectivity during the download, and the authentication step consumes a considerable amount of time, due to the DNS, TLS steps, etc...

Reply:
Seems reasonable.